tests/RobotFramework/tests/MQTT_health_check/mqtt_connection_status_with_incorrect_payload.robot

```diff
@@ -0,0 +1,25 @@
+*** Settings ***
+Resource    ../../resources/common.resource
+Library    ThinEdgeIO
+
+Test Tags    theme:troubleshooting    theme:monitoring    theme:c8y
+Suite Setup       Setup
+Suite Teardown    Get Logs
+
+
+*** Test Cases ***
+
+Send incorrect mqtt payload
+    Execute Command    tedge mqtt pub 'tedge/commands/req/software/list' '{"ids": "100"}'
+    ${output}=    Execute Command    systemctl status tedge-agent.service
+    Should Not Contain    ${output}    INFO mqtt_channel::connection: MQTT connection closed
```

Comment:
This might be a little too fragile, as the `systemctl status` command only returns the last ~10 lines. A better approach would be to use `journalctl` and its date filtering options, though there are a lot of potential pitfalls here (which they should be addressed by the following suggestion).

```suggestion
    ${start_time}=    ThinEdgeIO.Get Unix Timestamp
    Execute Command    tedge mqtt pub 'tedge/commands/req/software/list' '{"ids": "100"}'
    ${output}=    Execute Command    systemctl status tedge-agent.service
    Sleep    5s    reason=Allow sufficient time for the service to fail before checking the logs
    ${output}=    Execute Command    journalctl -u tedge-agent --since @${start_time} -n 1000
    Should Not Contain    ${output}    INFO mqtt_channel::connection: MQTT connection closed
```

It is still not ideal because using log entries in assertions will eventually break if we change the formatting of them, however due to a lack of any other option, we'll have to live with this method for now.

And this same change should be extend to the other test cases here.